crates/consensus/src/transaction/legacy.rs

```diff
@@ -226,7 +226,16 @@ impl Transaction for TxLegacy {
         if !header.list {
             return Err(alloy_rlp::Error::UnexpectedString);
         }
-        let tx = Self::decode_fields(buf)?;
+        let mut tx = Self::decode_fields(buf)?;
+
+        // If the buffer is not empty, it should be an eip-155 encoded chain_id.
+        if !buf.is_empty() {
+            let mut chain_buf = *buf;
+            let chain_id = ChainId::decode(&mut chain_buf)?;
+            // EIP-155: v = {0, 1} + CHAIN_ID * 2 + 35
+            tx.chain_id = ((chain_id - 35) >> 1).into();
+        }
+
         let signature = Signature::decode_rlp_vrs(buf)?;
```

Comment:
Shouldn't this be decoded from here? <https://github.com/gakonst/ethers-rs/blob/88095ba47eb6a3507f0db1767353b387b27a6e98/ethers-core/src/types/transaction/request.rs#L267>

Reply:
done in https://github.com/alloy-rs/alloy/pull/137/commits/a840f94dc372c4ce57837a260ee565879bdf333e